CHANGELOG.md

```diff
@@ -1,6 +1,97 @@
 # CHANGELOG
 
-## 0.3.4 @ next
+## 0.4.0 @ next
+
+- Switch from `node_modules` to Plug'n'Play (pnp) installations.
+
+  esy now uses the same approach as yarn (see [pnp rfc]) and installs all
+  package source code into a central source cache location.
+
+  `esy build` command now uses this source cache location to perform builds
+  from.
+
+  `_esy/default/pnp.js` is created with yarn's pnp runtime so that we keep
+  compatibility with JS ecosystem.
+
+  This result in much faster installation in case cache is warm and in less disk
+  space wasted on duplicated sources between the sandboxes.
+
+  To invoke npm installed binaries (`"bin"` field in `package.json`) one must
+  use:
+
+  ```
+  % esy webpack
+  ```
+
+  invocations. That means npm installed binaries are no in command environment
+  path.
+
+  To invoke `node` interpreter enhanced with pnp:
+
+  ```
+  % esy webpack
+  ```
+
+- New lock format.
+
+  esy 0.4.0 comes with a new lock format. Previously we stored everything in a
+  single JSON file `esy.lock.json`.
+
+  But as esy needs to store significantly more metadata about used packages
+  (opam metadata included and custom patches from overrides) single JSON file
+  isn't very good choice - it's big (over 9K lines usually), it's hard to review
+  during changes.
+
+  Instead of a single `esy.lock.json` file esy now produces `esy.lock` directory
+  with a JSON file `esy.lock/index.json` which keeps package graph and a set of
+  files, one for each opam/override/patch file store.
+
+- Added new command `esy show`.
+
+  Such command can be used to query metadata about npm, opam or other packages
+  (hosted on github for example).
+
+  Example:
+
+  ```
+  % esy show react
+  % esy show @opam/dune
+  % esy show github:facebook/reason
+  ```
+
+  Thanks to @kazcw for the feature.
+
+- Numerous fixes and improvements for Windows.
```

Comment:
We should call out the work @prometheansacrifice did here too for the native esy-bash, which is now integrated with `esy`. Maybe even link to the PR if it makes sense: https://github.com/bryphe/esy-bash/pull/19

Reply:
Also the work @ulrikstrid did to enable `esy release` on Windows: https://github.com/esy/esy/pull/519

Reply:
Yes, I remember, going through the `git log`!